### PR TITLE
ambient: more strict UT check for event handler

### PR DESCRIPTION
**Pull request by PlatformLC**
**Please provide a description of this PR:**

add more strict UT check for event handler

### CODE REVIEW

**Review by hzxuzhonghu — APPROVED** [review]

File: cni/pkg/nodeagent/informers_test.go [hunk 1]
```diff
@@ -384,6 +384,16 @@ func TestExistingPodRemovedWhenPodAnnotated(t *testing.T) {
 
 	assertPodNotAnnotated(t, client, pod)
 
+	// path a test label to emulate a non-annotation POD update event
```
Comment (bleggett) [inline cni/pkg/nodeagent/informers_test.go:387]:
```suggestion
	// patch a test label to emulate a non-annotation POD update event
```

File: cni/pkg/nodeagent/informers_test.go [hunk 2]
```diff
@@ -384,6 +384,16 @@ func TestExistingPodRemovedWhenPodAnnotated(t *testing.T) {
...
+	_, err = client.Kube().CoreV1().Pods(pod.Namespace).Patch(ctx, pod.Name,
+		types.MergePatchType, []byte(`{"metadata":{"labels":{"test":"update"}}}`), metav1.PatchOptions{})
+	assert.NoError(t, err)
+
+	// wait for an update events
+	mt.Assert(EventTotals.Name(), map[string]string{"type": "update"}, monitortest.AtLeast(5))
```
Comment (bleggett) [inline cni/pkg/nodeagent/informers_test.go:393]: Aha, now I see. Good catch. This should be fine in practice as the removal will be a no-op but either way, it is good to have a test for this.

**Review by bleggett — APPROVED** [review]
Thanks for following up - now that I see the test case I understand what you're getting at. 

With the test in place I am good to merge this when CI passes.

**Pull request merged**